addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -156,9 +156,62 @@ export class PosStore extends Reactive {
     async initServerData() {
         await this.processServerData();
         this.onNotified = getOnNotified(this.bus, this.config.access_token);
+        this.onNotified("CLOSING_SESSION", this.closingSessionNotification.bind(this));
+        this.onNotified("CANCEL_ORDERS", this.cancelOrderNotification.bind(this));
+
         return await this.afterProcessServerData();
     }
 
+    async cancelOrderNotification(data) {
+        if (data.login_number === this.session.login_number) {
+            return;
+        }
+
+        const orders = this.models["pos.order"].readMany(data.order_ids);
+        for (const order of orders) {
+            if (!order.finalized) {
+                order.state = "cancel";
+            }
+        }
+    }
+
+    async closingSessionNotification(data) {
+        if (data.login_number === this.session.login_number) {
+            return;
+        }
+
+        try {
+            const paidOrderNotSynced = this.models["pos.order"].filter(
+                (order) => order.state === "paid" && order.id !== "number"
+            );
+            this.addPendingOrder(paidOrderNotSynced.map((o) => o.id));
+            await this.syncAllOrders({ throw: true });
+
+            this.dialog.add(AlertDialog, {
+                title: _t("Closing Session"),
+                body: _t("The session is being closed by another user. The page will be reloaded."),
+            });
+        } catch {
+            this.dialog.add(AlertDialog, {
+                title: _t("Error"),
+                body: _t(
+                    "An error occurred while closing the session. Unsynced orders will be available in the next session."
+                ),
+            });
+        } finally {
+            const orders = this.models["pos.order"].filter((o) => o.id !== "number");
+            for (const order of orders) {
+                if (!order.finalized) {
+                    order.state = "cancel";
+                }
+            }
+        }
+
+        setTimeout(() => {
+            window.location.reload();
+        }, 3000);
```

Comment:
Shouldn't we show a popup saying "The session is being closed by another user. You should reload the page to ensure data integrity" without forcing the reload ? What if someone is doing an order and lose it ? And also maybe add some text to the user closing a session saying: if someone is still doing orders, you could have some data discrepancy or something like that ?  What do you think ?

Reply:
Before reloading if there is an order that isn't sync with the server, it will be kept in indexeddb and sync with the next session